dotCMS/src/main/java/com/dotcms/publisher/bundle/business/BundleAPIImpl.java

```diff
@@ -82,7 +107,180 @@ public Bundle getBundleById(String id) throws DotDataException {
+
+		/**
+		if (!APILocator.getPermissionAPI().doesUserHavePermission(bundle,
+				PermissionAPI.PERMISSION_EDIT, user, false)) {
+
+			throw new DotSecurityException("User : " + user.getUserId() +
+					", is not allowed to delete the bundle: " + bundleId)
+		}*/
+		// todo: bundle is not a permissionable yet, so can not validate yet.
+	}
+
+	@WrapInTransaction
+	@Override
+	public Set<String> deleteBundleAndDependenciesOlderThan(final Date olderThan, final User user) throws DotDataException {
+
+		final ImmutableSet.Builder<String> bundlesDeleted = new ImmutableSet.Builder<>();
+		final int limit          = 100;
+		int offset               = 0;
+		final boolean isAdmin    = this.userAPI.isCMSAdmin(user);
+		List<Bundle> sentBundles = isAdmin?
+				this.bundleFactory.findSentBundles(olderThan, limit, offset):
+				this.bundleFactory.findSentBundles(olderThan, user.getUserId(), limit, offset);
+
+		Logger.info(this, "Deleting bundles older than: " + olderThan);
+		while (UtilMethods.isSet(sentBundles)) {
+
+			for (final Bundle bundle : sentBundles) {
+
+				this.deleteBundleAndDependencies(bundle, user);
+				bundlesDeleted.add(bundle.getId());
+			}
+
+			sentBundles = isAdmin?
+					this.bundleFactory.findSentBundles(olderThan, limit, offset):
+					this.bundleFactory.findSentBundles(olderThan, user.getUserId(), limit, offset);
+		}
+
+		return bundlesDeleted.build();
+	}
+
+	@WrapInTransaction
+	@Override
+	public Set<String>  deleteAllBundles(final User user,
+										 final PublishAuditStatus.Status ...statuses) throws DotDataException {
+
+		final ImmutableSet.Builder<String> bundlesDeleted = new ImmutableSet.Builder<>();
+		final int limit          = 100;
+		final int offset         = 0;
+		final boolean isAdmin    = this.userAPI.isCMSAdmin(user);
+		List<Bundle> sentBundles = isAdmin?
+				this.bundleFactory.findSentBundles(limit, offset): // todo: modify this method in order to use publishing_queue_audit,
+				this.bundleFactory.findSentBundles(user.getUserId(), limit, offset);
+
+		Logger.info(this, "Deleting all bundles with statuses: "
+				+ Arrays.asList(statuses));
+		while (UtilMethods.isSet(sentBundles)) {
+
+			for (final Bundle bundle : sentBundles) {
+
+				try {
+
+					final PublishAuditStatus status =
+							this.publishAuditAPI.getPublishAuditStatus(bundle.getId());
+
+					if (this.containsStatus (status, statuses)) {
+
+						this.deleteBundleAndDependencies(bundle, user);
+						bundlesDeleted.add(bundle.getId());
+					}
+				} catch (final DotPublisherException e) {
+
+					throw new DotDataException(e);
+				}
+			}
+
+			sentBundles = isAdmin?
+					this.bundleFactory.findSentBundles(limit, offset):
+					this.bundleFactory.findSentBundles(user.getUserId(), limit, offset);
+		}
+
+		return bundlesDeleted.build();
+	} // deleteAllBundles.
+
+	private boolean containsStatus(final PublishAuditStatus          bundleStatus,
+								   final PublishAuditStatus.Status[] statuses) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Consider using varargs for methods or constructors which take an array the last parameter.](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4502981)